src/Npgsql/TypeMapping/TypeMapperBase.cs

```diff
@@ -79,32 +80,43 @@ public bool UnmapEnum<TEnum>(string? pgName = null, INpgsqlNameTranslator? nameT
         #region Composite mapping
 
         public INpgsqlTypeMapper MapComposite<T>(string? pgName = null, INpgsqlNameTranslator? nameTranslator = null)
-        {
+            => MapComposite(typeof(T), pgName, nameTranslator);
+
+        public bool UnmapComposite<T>(string? pgName = null, INpgsqlNameTranslator? nameTranslator = null)
+            => UnmapComposite(typeof(T), pgName, nameTranslator);
+
+        public INpgsqlTypeMapper MapComposite(Type compType, string? pgName = null, INpgsqlNameTranslator? nameTranslator = null) {
             if (pgName != null && pgName.Trim() == "")
                 throw new ArgumentException("pgName can't be empty", nameof(pgName));
 
             if (nameTranslator == null)
                 nameTranslator = DefaultNameTranslator;
             if (pgName == null)
-                pgName = GetPgName<T>(nameTranslator);
+                pgName = GetPgName(compType, nameTranslator);
 
-            return AddMapping(new NpgsqlTypeMappingBuilder
-            {
+            var thfType = typeof(CompositeTypeHandlerFactory<>);
+            var thf = (NpgsqlTypeHandlerFactory)Activator.CreateInstance(
```

Comment:
This requires MakeGenericType even in cases where the user uses the generic overload, which is bad for AOT scenarios. In general, we're trying to minimize/remove cases of code generation as much as possible.

At the very least, I think that for the generic path (`MapComposite<T>`) we should do MakeGenericType. This can be done by doing the opposite - create a generic method from the non-generic path.

Reply:
Let's merge as is. I have some ideas on it already (:

Reply:
Any other issues?

Reply:
No other issues really, but I'd prefer if we didn't merge something where we know there's a problem... If you want to fix this right away then sure.

Reply:
Right now Npgsql and AOT are not friends, you know it. So this PR is fine in the current circumstances (:

Reply:
I don't want us to needlessly make the situation worse though... It's very easy to flip this around and only use code generation when the "bad" overload is used.